account_budget_activity/models/account_budget.py

```diff
@@ -702,48 +704,37 @@ def _compute_planned_amount(self):
             rec.planned_amount = planned_amount  # from last year
 
     @api.multi
-    @api.depends('r1', 'r2', 'r3', 'r4', 'r5', 'r6',
-                 'r7', 'r8', 'r9', 'r10', 'r11', 'r12', )
+    @api.depends('released')
     def _compute_released_amount(self):
         for rec in self:
-            released_amount = sum([(rec.m1 * rec.r1), (rec.m2 * rec.r2),
-                                   (rec.m3 * rec.r3), (rec.m4 * rec.r4),
-                                   (rec.m5 * rec.r5), (rec.m6 * rec.r6),
-                                   (rec.m7 * rec.r7), (rec.m8 * rec.r8),
-                                   (rec.m9 * rec.r9), (rec.m10 * rec.r10),
-                                   (rec.m11 * rec.r11), (rec.m12 * rec.r12),
-                                   ])
+            released_amount = sum([rec.m1, rec.m2, rec.m3, rec.m4,
+                                  rec.m5, rec.m6, rec.m7, rec.m8,
+                                  rec.m9, rec.m10, rec.m11, rec.m12])
             rec.released_amount = released_amount  # from last year
 
     @api.multi
     def release_budget_line(self, release_result):
-        # Always release the former period. I.e, r6 will include r1-r5
-        periods = list({int(k[1:]): v
-                        for k, v in release_result.iteritems()
-                        if v})
-        period = periods and max(periods) or 0
-        vals = {'r'+str(i+1): (i+1) <= period for i in range(12)}
         for rec in self:
-            old_vals = {'r'+str(i+1): rec['r'+str(i+1)] for i in range(12)}
-            if old_vals != vals:
-                rec.write(vals)
+            amount_to_release = release_result.get(rec.id, 0.0)
+            if amount_to_release > rec.planned_amount - rec.released_amount:
+                raise UserError(_("You don't have enough amount to release!"))
+            new_release_amount = rec.released_amount + amount_to_release
+            rec.write({'released_amount': new_release_amount})
         return
 
     @api.multi
     def _prepare_budget_line_release(self, periods):
         self.ensure_one()
         releases = []
-        if not periods:
-            return []
         for period in periods:
             from_period = period[0]
             to_period = period[-1]
-            release = self['r'+str(to_period)]
+            release = 12
```

Comment:
Not sure what this is for? As we have no concept of 12 months, do we still need it?